index.js

```diff
@@ -14,4 +14,13 @@
  * limitations under the License.
  */
 
-module.exports = require('./lib/Puppeteer');
+// If node does not support async await, use the compiled version.
+let folder = 'lib';
+try {
+  new Function('async function test(){await 1}');
+}
+catch (error) {
+  folder = 'node6';
```

Comment:
Not sure if there's a version check somewhere, but it might be nice to make sure they have node 6 in this fallback case, otherwise they'll get strange errors with generators (?)

```
process.version
```